# **Request to PUNK IPA API AND Dataset**

# Reading the Dataset of Beers

In [1]:

import pandas as pd
beers_new = pd.read_csv('Raw Data/raw_Data/beers.csv')

#First, get the columns we want:

keep_col = ['brewery_id','name','cat_id','style_id','abv',]
beers_new = beers_new[keep_col]
beers_new


#Now, by having a look inside , we have to  eliminate the rows with '-1' in categories because there does not exist  
#a category with the id '-1' and it has two types: integer and string.


beers_new= beers_new[~(beers_new.cat_id == "-1")]
beers_new = beers_new[~(beers_new.cat_id == -1 )]



#'''eliminating the rows with '0.0' in abv because all the beers have al least some alcohol inside and 
#drop the descript column because we are not going to use it and it has two types: integer and string.'''

beers_new = beers_new[~(beers_new.abv=="0.0")]
beers_new = beers_new[~(beers_new.abv==0.0)]



beers_new  

/usr/local/Cellar/ipython/7.8.0/libexec/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3058: DtypeWarning: Columns (0,1,3,4,5,6,7,8,9,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,41,46,48,49,50,51,52,54,58,60,61,62,63,64,66,70,71,72,73,74,75,76,78,82,84,85,86,87,88,90,94,96,97,98,99,100,102,108,109,110,111,112,114,118,119,121,122,123,124,126,131,133,134,135,136,138,143,145,146,147,148,150,155,157,158,159,160,162,167,169,170,171,172,174,179,181,182,183,184,186,191,193,194,195,196,198,203,205,206,207,215,217,219,227,229,231,239,240,241,242,243,245,247,251,253,255,257,263,264,266,268,276,278,280,288,290,292,300,302,304,312,314,316,324,326,328,336,338,340,348,349,350,351,353,355,363,364,366,368,376,378,380,388,390,392,400,402,404,412,414,416,424,426,428,436,438,440,448,450,452,460,462,464,474,476,484,486,488,496,498,500,508,510,512,520,522,524,532,534,536,544) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=int

brewery_id                               name cat_id style_id  abv
0           812                        Hocus Pocus     11      116  4.5
5          1385                      Winter Warmer      1       13  5.2
7          1099                      Oatmeal Stout      3       42    5
13         1099                     Imperial Stout      3       42    7
16         1023                 Wiesen Edel Weisse      4       50  6.2
...         ...                                ...    ...      ...  ...
5843       1417                      Watershed IPA      3       30  7.1
5844       1087          Estrella Levante Especial      9      106  5.4
5845       1087  Estrella Levante Sin 0.0% Alcohol      9      106    1
5846       1419            Koshihikari Echigo Beer     11      132    5
5848       1421                         The Public      3       26    6

[2568 rows x 5 columns]

### Let's look for the nan values  

In [2]:
beers_new.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2568 entries, 0 to 5848
Data columns (total 5 columns):
brewery_id    2544 non-null object
name          2544 non-null object
cat_id        2539 non-null object
style_id      2536 non-null object
abv           2532 non-null object
dtypes: object(5)
memory usage: 120.4+ KB


### There some nan value inside the dataset, so we are going to eliminate them because is impossible to  vinculate the missing values with all the data we have.

In [3]:
beers_new = beers_new.dropna()
beers_new.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2532 entries, 0 to 5848
Data columns (total 5 columns):
brewery_id    2532 non-null object
name          2532 non-null object
cat_id        2532 non-null object
style_id      2532 non-null object
abv           2532 non-null object
dtypes: object(5)
memory usage: 118.7+ KB


## Let's look inside the column abv

### Looking inside the dataset, we found that some values in the abv column are words which we are going to delete them.

In [4]:
beers_new.abv.unique()

array([4.5, 5.2, 5.0, 7.0, 6.2, 8.7, 5.9, 5.6, 4.8, 8.0, 7.5, 6.0, 5.1,
       10.0, 5.5, 8.5, 6.7, 6.6, 6.5, 5.94, 6.3, 11.5, 9.7, 6.1, 5.4, 6.4,
       4.4, 11.7, 5.8, 10.5, 8.3, 8.6, 16.03, 9.28, 11.6, 4.9, 5.3, 7.7,
       4.2, 8.1, 3.9, 4.6, 9.0, 6.8, 7.25, 4.3, 3.8, 3.5, 7.3, 5.7, 8.9,
       11.3, 9.2, 7.1, 11.0, 14.0, 12.0, 8.2, 7.2, 9.1, 4.0, 9.5, 4.7,
       7.4, 6.45, 5.75, 9.9, 10.2, 15.04, 9.6, 6.9, 7.8, 11.4, 12.7, 4.1,
       14.9, 7.85, 4.73, 6.54, 5.57, 4.99, 4.95, 8.75, 6.15, 11.9, 9.4,
       15.0, 8.41, 7.9, 7.6, 5.86, 4.85, 10.1, 99.99, 9.47, 3.6, 13.0,
       11.1, 5.52, 3.3, 3.4, 4.97, 6.75, 3.75, 4.66, 4.74, 4.19, 3.79,
       4.15, 4.75, 8.8, 4.94, 9.3, 11.8, 18.0, 5.25, 9.8, 3.2, 5.35,
       '10.4', '7', '0', '5', '5.7', '8.5', '9', '8', '4.5', '6', '7.6',
       '4.7', '5.2', '5.4', '10.5', '6.5', '5.3', '6.2', '5.6', '5.8',
       '10.2', '4.8', '6.1', '7.7', '10', '7.5', '5.5', '12', '4.6',
       '6.9', '10.8', '7.8', '4.4', '5.1', '9.5', '7.4', ' Chinook

### All wrong values are  going to be change to NaN and we are going to drop them too.

In [55]:
beers_new.abv = pd.to_numeric(beers_new.abv, errors='coerce')
beers_new.abv.unique()

array([ 4.5 ,  5.2 ,  5.  ,  7.  ,  6.2 ,  8.7 ,  5.9 ,  5.6 ,  4.8 ,
        8.  ,  7.5 ,  6.  ,  5.1 , 10.  ,  5.5 ,  8.5 ,  6.7 ,  6.6 ,
        6.5 ,  5.94,  6.3 , 11.5 ,  9.7 ,  6.1 ,  5.4 ,  6.4 ,  4.4 ,
       11.7 ,  5.8 , 10.5 ,  8.3 ,  8.6 , 16.03,  9.28, 11.6 ,  4.9 ,
        5.3 ,  7.7 ,  4.2 ,  8.1 ,  3.9 ,  4.6 ,  9.  ,  6.8 ,  7.25,
        4.3 ,  3.8 ,  3.5 ,  7.3 ,  5.7 ,  8.9 , 11.3 ,  9.2 ,  7.1 ,
       11.  , 14.  , 12.  ,  8.2 ,  7.2 ,  9.1 ,  4.  ,  9.5 ,  4.7 ,
        7.4 ,  6.45,  5.75,  9.9 , 10.2 , 15.04,  9.6 ,  6.9 ,  7.8 ,
       11.4 , 12.7 ,  4.1 , 14.9 ,  7.85,  4.73,  6.54,  5.57,  4.99,
        4.95,  8.75,  6.15, 11.9 ,  9.4 , 15.  ,  8.41,  7.9 ,  7.6 ,
        5.86,  4.85, 10.1 , 99.99,  9.47,  3.6 , 13.  , 11.1 ,  5.52,
        3.3 ,  3.4 ,  4.97,  6.75,  3.75,  4.66,  4.74,  4.19,  3.79,
        4.15,  4.75,  8.8 ,  4.94,  9.3 , 11.8 , 18.  ,  5.25,  9.8 ,
        3.2 ,  5.35, 10.4 ,  0.  , 10.8 ,  8.4 ,  3.85,  4.65, 20.  ,
       13.6 ,  6.85,

### I drop the last nan values.

In [19]:
#I drop the nan value from before:
beers_new = beers_new.dropna()


<class 'pandas.core.frame.DataFrame'>
Int64Index: 2523 entries, 0 to 2522
Data columns (total 5 columns):
brewery_id    2523 non-null object
name          2523 non-null object
cat_id        2523 non-null object
style_id      2523 non-null object
abv           2523 non-null float64
dtypes: float64(1), object(4)
memory usage: 118.3+ KB


brewery_id                               name cat_id style_id  abv
0           812                        Hocus Pocus     11      116  4.5
1          1385                      Winter Warmer      1       13  5.2
2          1099                      Oatmeal Stout      3       42  5.0
3          1099                     Imperial Stout      3       42  7.0
4          1023                 Wiesen Edel Weisse      4       50  6.2
...         ...                                ...    ...      ...  ...
2518       1417                      Watershed IPA      3       30  7.1
2519       1087          Estrella Levante Especial      9      106  5.4
2520       1087  Estrella Levante Sin 0.0% Alcohol      9      106  1.0
2521       1419            Koshihikari Echigo Beer     11      132  5.0
2522       1421                         The Public      3       26  6.0

[2523 rows x 5 columns]

### Drop 3 more values which are number in the column 'Name' because we can not identify the real meaning or beer with the information we have.

In [ ]:
#Checking the values if they have sense:
beers_new.cat_id.unique()

# Locate the number and drop them from the column:
beers_new = beers_new.drop(beers_new[beers_new.name.str.isdigit()].index).reset_index(drop=True)

beers_new

#### Eliminate the value which has none comment sense of abv, which is the one with 99.99

In [65]:
beers_new[(beers_new.abv == beers_new.abv.max())]
beers_new = beers_new.drop(565)


Empty DataFrame
Columns: [brewery_id, name, cat_id, style_id, abv]
Index: []

### All the values change correctly to float number, so we are going to see the other columns

In [33]:
beers_new.style_id.unique()    # we can change them to floating numbers, nothing rare

beers_new.cat_id.unique()      # we can change them to floating numbers, nothing rare

beers_new.brewery_id.unique() #we can change them to floating numbers, nothing rare



array([812, 1385, 1099, 1023, 483, 1220, 878, 935, 1307, 723, 160, 1221,
       670, 923, 1142, 161, 779, 266, 1202, 1185, 1260, 1056, 884, 345,
       727, 1075, 304, 481, 441, 584, 397, 562, 622, 447, 1392, 919, 1084,
       595, 31, 307, 859, 39, 810, 62, 1136, 765, 1315, 484, 905, 70, 604,
       296, 708, 590, 1179, 1274, 312, 624, 1126, 701, 892, 588, 385, 308,
       1004, 707, 133, 236, 272, 285, 7, 167, 903, 431, 757, 1115, 363,
       425, 609, 586, 98, 844, 156, 127, 367, 1375, 638, 1043, 744, 626,
       1204, 262, 511, 230, 1120, 74, 204, 44, 233, 1116, 1344, 216, 357,
       873, 598, 457, 829, 496, 497, 1296, 842, 445, 1124, 1151, 504, 526,
       1030, 494, 641, 1177, 21, 540, 459, 221, 368, 745, 301, 169, 671,
       370, 1303, 742, 175, 1331, 803, 26, 691, 823, 503, 916, 283, 1321,
       1158, 409, 841, 700, 549, 71, 612, 15, 1181, 527, 1169, 703, 1242,
       1366, 592, 371, 36, 491, 1258, 866, 601, 1027, 1129, 1305, 76, 144,
       537, 263, 1379, 440, 1045, 1243, 

## All the columns do not have any strange number or text inside, so we are going to pocess to change the type of them.

In [35]:
beers_new[['brewery_id','abv', 'cat_id', 'style_id']] = beers_new[['brewery_id','abv', 'cat_id', 'style_id']].astype('float') 
beers_new.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2523 entries, 0 to 2522
Data columns (total 5 columns):
brewery_id    2523 non-null float64
name          2523 non-null object
cat_id        2523 non-null float64
style_id      2523 non-null float64
abv           2523 non-null float64
dtypes: float64(4), object(1)
memory usage: 98.7+ KB


In [36]:
beers_new

brewery_id                               name  cat_id  style_id  abv
0          812.0                        Hocus Pocus    11.0     116.0  4.5
1         1385.0                      Winter Warmer     1.0      13.0  5.2
2         1099.0                      Oatmeal Stout     3.0      42.0  5.0
3         1099.0                     Imperial Stout     3.0      42.0  7.0
4         1023.0                 Wiesen Edel Weisse     4.0      50.0  6.2
...          ...                                ...     ...       ...  ...
2518      1417.0                      Watershed IPA     3.0      30.0  7.1
2519      1087.0          Estrella Levante Especial     9.0     106.0  5.4
2520      1087.0  Estrella Levante Sin 0.0% Alcohol     9.0     106.0  1.0
2521      1419.0            Koshihikari Echigo Beer    11.0     132.0  5.0
2522      1421.0                         The Public     3.0      26.0  6.0

[2523 rows x 5 columns]

##### Create the new dataset for beers which we are going to use.

In [39]:
beers_new.to_csv('NEW_DATA/beers_new.csv',index=False)

#### Read the file from beers:

In [40]:
beers = pd.read_csv('NEW_DATA/beers_new.csv')


# Import table of breweries and create the new dataset of it :

In [42]:
breweries = pd.read_csv('Raw Data/Old_Data/breweries.csv')


breweries = breweries.drop( columns= ['filepath','phone','website','last_mod','code','address2','descript'] )

#checking if there is any problem with the data:
breweries.info()




<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1414 entries, 0 to 1413
Data columns (total 6 columns):
id          1414 non-null int64
name        1414 non-null object
address1    1145 non-null object
city        1392 non-null object
state       1232 non-null object
country     1413 non-null object
dtypes: int64(1), object(5)
memory usage: 66.4+ KB


# Looking for any problem inside the columns:


In [45]:
breweries.isna().sum()

# We can let this Nan values because we do not need it for the datasets analysis.


id            0
name          0
address1    269
city         22
state       182
country       1
dtype: int64

In [ ]:
#creating the new file of breweries:

breweries.to_csv('REAL_Data_beers/NEW_DATA/breweries_new.csv',index=False)

#for reading the file:

breweries = pd.read_csv('NEW_DATA/breweries_new.csv',index=False)

# Import table of categories

In [49]:
categories = pd.read_csv('Raw Data/Old_Data/categories.csv')

#drop column not using:
categories = categories.drop(columns=['last_mod'])

# Create the new csv:
categories.to_csv('NEW_DATA/categories_new.csv',index=False)


categories.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11 entries, 0 to 10
Data columns (total 2 columns):
id          11 non-null int64
cat_name    11 non-null object
dtypes: int64(1), object(1)
memory usage: 304.0+ bytes


# Import table from styles and droping a column.

In [54]:
styles = pd.read_csv('Raw Data/Old_Data/styles.csv')


styles = styles.drop(columns=['last_mod'])

styles.to_csv('NEW_DATA/Style_new.csv',index=False)